src/libraries/Microsoft.PowerFx.Core/Binding/Binder.cs

```diff
@@ -4983,7 +5002,26 @@ public override bool PreVisit(CallNode node)
                 _currentScope = scopeNew.Parent;
                 PostVisit(node.Args);
 
-                // Typecheck the invocation.
+                if (maybeFunc.HasIdentifiers && _features.HasFlag(Features.SupportIdentifiers))
+                {
+                    var i = 0;
+
+                    foreach (var arg in args)
+                    {
+                        i++;
+
+                        if (arg is FirstNameNode firstNameNode && maybeFunc.IsIdentifierParam(i))
+                        {
+                            if (DType.TryGetLogicalNameForColumn(argTypes[0], firstNameNode.Ident.Name, out var logicalName))
+                            {
```

Comment:
if you use:
GetLogicalNodeNameAndUpdateDisplayNames(argTypes[0], firstNameNode.Iden), this will also handle the display name rewrite cases.

Reply:
Am now using it but don't understand what it's really doing and how it's different from the one I used.